src/containers/edit-profile/password-security-tab/PasswordSecurityTab.tsx

```diff
@@ -39,48 +138,56 @@ const PasswordSecurityTab = () => {
         style={styles.titleAndDescription}
         title={t('editProfilePage.profile.passwordSecurityTab.title')}
       />
-      <Box>
+      <Box component='form' onSubmit={handleSubmit}>
         <Typography sx={styles.subtitle}>
           {t('editProfilePage.profile.passwordSecurityTab.changePassword')}
         </Typography>
-        <AppTextField
-          fullWidth
-          label={t(
-            'editProfilePage.profile.passwordSecurityTab.currentPassword'
-          )}
-          onChange={handleInputChange('currentPassword')}
-          type={InputEnum.Password}
-          value={data.currentPassword}
-        />
-        <AppTextField
-          errorMsg={t(errors.password)}
-          fullWidth
-          label={t('editProfilePage.profile.passwordSecurityTab.newPassword')}
-          onBlur={handleBlur('password')}
-          onChange={handleInputChange('password')}
-          type={InputEnum.Password}
-          value={data.password}
-        />
-        <AppTextField
-          errorMsg={t(errors.confirmPassword)}
-          fullWidth
-          label={t(
-            'editProfilePage.profile.passwordSecurityTab.retypePassword'
-          )}
-          onBlur={handleBlur('confirmPassword')}
-          onChange={handleInputChange('confirmPassword')}
-          type={InputEnum.Password}
-          value={data.confirmPassword}
-        />
+        <Box sx={styles.form}>
+          <AppTextField
+            InputProps={currentPasswordVisibility}
+            errorMsg={t(errors.currentPassword)}
+            fullWidth
+            label={t(
+              'editProfilePage.profile.passwordSecurityTab.currentPassword'
+            )}
+            onBlur={handleBlur('currentPassword')}
+            onChange={handleCurrentPasswordChange}
+            type={showCurrentPassword ? InputEnum.Text : InputEnum.Password}
```

Comment:
As I see it's the same logic below for `showPassword` and here with `showCurrentPassword`.
Can define `function` to avoid duplication and use it in your case

Reply:
Thank you! Fixed.